.travis.yml

```diff
@@ -54,19 +56,36 @@ matrix:
       addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-7.4.2","cabal-install-2.4","freeglut3-dev"]}}
     - compiler: ghc-head
       addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-head","cabal-install-head","freeglut3-dev"]}}
-      env: GHCHEAD=true
   allow_failures:
     - compiler: ghc-head
     - compiler: ghc-8.8.1
 before_install:
+  - |
+    if [ "$TRAVIS_OS_NAME" = "linux" ]; then
+    sudo add-apt-repository -y ppa:hvr/ghc;
+    sudo add-apt-repository -y ppa:hvr/ghcjs;
+    sudo apt-get update;
+    sudo apt-get install $CC cabal-install-3.0 freeglut3-dev;
```

Comment:
Why is this part necessary, given that `cabal-install-3.0` and `freeglut3-dev` are both listed in the `addons` part of the configuration matrix?

Reply:
With bionic apt pragma isn’t executed. No idea why.

Reply:
Reverted to `xenial`. I'm not sure even where to report anything to travis nowadays. And especially about something which "accidentally" works, not publicly announced to be even an experimental feature.